src/main/kotlin/graphics/scenery/primitives/Atmosphere.kt

```diff
@@ -12,56 +12,73 @@ import java.lang.Math.toRadians
 import java.time.LocalDateTime
 import kotlin.collections.HashMap
 import kotlin.math.*
+import kotlin.time.Duration.Companion.seconds
 
 /**
  * Implementation of a Nishita sky shader, applied to an [Icosphere] that wraps around the scene as a skybox.
  * The shader code is ported from Rye Terrells [repository](https://github.com/wwwtyro/glsl-atmosphere).
  * To move the sun with arrow keybinds, attach the behaviours using the [attachRotateBehaviours] function.
- * @param initSunDir [Vector3f] of the sun position. Defaults to sun elevation of the current local time.
- * @param emissionStrength Emission strength of the atmosphere shader. Defaults to 0.3f.
+ * @param initSunDirection [Vector3f] of the sun position. Defaults to sun elevation of the current local time.
+ * @param emissionStrength Emission strength of the atmosphere shader. Defaults to 1f.
  * @param latitude Latitude of the user; needed to calculate the local sun position. Defaults to 50.0, which is central Germany.
  */
-open class Atmosphere(initSunDir: Vector3f? = null, emissionStrength: Float = 0.3f, var latitude: Double = 50.0) :
+open class Atmosphere(
+    initSunDirection: Vector3f? = null,
+    emissionStrength: Float = 1.0f,
+    var latitude: Float = 50.0f
+) :
     Icosphere(10f, 2, insideNormals = true) {
 
     @ShaderProperty
-    var sunDir: Vector3f
+    var sunDirection = Vector3f(1f, 1f, 1f)
 
+    /** Is set to true if the user manually moved the sun direction. This disables automatic updating.*/
     private var sunDirectionManual: Boolean = false
+    /** Flag that tracks whether the sun position controls are currently attached to the input handler. */
+    var hasControls: Boolean = false
+        private set
+    // Coroutine job for updating the sun direction
+    private var job = CoroutineScope(Dispatchers.Default).launch(start = CoroutineStart.LAZY) {
+        logger.debug("Launched sun updating job")
+        while (this.coroutineContext.isActive) {
```

Comment:
I think this is wrong, you're checking whether the context is active, not the job. It should just be `isActive`, iirc.

Reply:
good point, thanks